src/Common/src/Interop/Windows/SspiCli/Interop.SSPI.cs

```diff
@@ -25,6 +26,7 @@ internal struct CredHandle
 
             public bool IsZero
             {
+                [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Why is this needed?

Reply:
Doesn't inline

Reply:
> Doesn't inline

I'm surprised that `return dwLower == IntPtr.Zero && dwUpper == IntPtr.Zero;` wouldn't inline.  What was the reasoning it gave?
cc: @AndyAyersMS

Reply:
The reason chain is
```
Inlines into 0600025A SslStream:get_IsAuthenticated():bool:this
  [0 IL=0014 TR=000011 06000324] [FAILED: unprofitable inline] SecureChannel:get_IsValidContext():bool:this
```
  
```
Inlines into 06000324 SecureChannel:get_IsValidContext():bool:this
  [0 IL=0014 TR=000011 0600041E] [FAILED: unprofitable inline devirt] SafeDeleteContext:get_IsInvalid():bool:this
```

```
Inlines into 0600041E SafeDeleteContext:get_IsInvalid():bool:this
  [1 IL=0001 TR=000001 0600327E] [below ALWAYS_INLINE size] SafeHandle:get_IsClosed():bool:this
  [0 IL=0014 TR=000014 0600001A] [FAILED: unprofitable inline] CredHandle:get_IsZero():bool:this
```

```
Inlines into 0600001A CredHandle:get_IsZero():bool:this
  [1 IL=0011 TR=000003 06001386] [profitable inline] IntPtr:op_Equality(long,long):bool
  [2 IL=0029 TR=000018 06001386] [profitable inline] IntPtr:op_Equality(long,long):bool
```

Reply:
It's likely the extra apparent complexity from the calls to `operator==`. 

The JIT's profitability assessment for inlining `IsZero` is entirely based on examination of the IL stream. If there are calls in the IL stream the jit has no idea what methods might be called, and in general it is too expensive (at jit time) to ask the runtime to resolve the tokens for calls that an prospective inline candidate method makes. 

I don't expect this state of affairs to change anytime soon, but it is possible.
* I have explored updating the IL based heuristics and remain convinced they can be improved, but it is not going to be easy. 
* There is some hope that with tiering the jit can afford to do a bit more in-depth exploration of inline candidates at Tier1. I don't think we are there quite yet. 
* Perhaps the jit can spend more time doing analysis when prejitting and leave hints to itself that it can consume when jitting.

But for now, in cases like this where those next-level calls are "known" by the programmer to produce compact code, some hinting is needed to get `IsZero` to inline.